src/components/composer/composer/composer.ts

```diff
@@ -309,8 +309,8 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
     } else {
       ev.stopPropagation();
     }
-    const { start, end } = this.contentHelper.getCurrentSelection();
-    if (!this.env.model.getters.isSelectingForComposer()) {
+    if (!this.env.model.getters.isSelectingForComposer() && !["Enter", "Tab"].includes(ev.key)) {
```

Comment:
What I don't like about the proposed solution here is that handlers business are leaking in the generic `onKeydown`.
I'm wondering if the selection changed shouldn't be moved inside the `else`.
And managed manually by the handlers if they need to (probably only arrow keys, and they are already, to a certain extend)

Reply:
might be a better solution, yup

Reply:
Are you referring to moving the external condition into the if condition? If so, then it seems to work on UI but test cases are not liking it. Second approach I tried was to remove the external condition completely, and that also did solve the issue but it breaks the test cases as well.

So I just want to get a more clear idea why do we have the external condition? Handler can take care of the code with desired behavior quiet well, yet we have an dispatch outside to set the cursor place. (Yes I have referred the commit which brought the change but got confused a bit)

Reply:
I guess now I got the idea behind it, I have made some changes to my code according to your suggestion. Hopefully this will serve as required. Let me know if any changes are required / if I misunderstand something.